snapcraft/parts/plugins/colcon_plugin.py

```diff
@@ -197,6 +197,12 @@ def _get_build_commands(self) -> List[str]:
         if options.colcon_packages:
             build_command.extend(["--packages-select", *options.colcon_packages])
 
+        # compile in release only if user did not set cmake_args or cmake build type in cmake_args
+        if not options.colcon_cmake_args or not any(
+            "-DCMAKE_BUILD_TYPE=" in s for s in options.colcon_cmake_args
+        ):
```

Comment:
Isn't `any` going to return false anyway when options.colcon_cmake_args is empty? Do you really need the first part of the `if`

Reply:
The first part of the if may make sure that the variable isn't `None`, altho I'm not sure whether it can take this value.